From e4fd7748349509519ee4c7146697397937020d3f Mon Sep 17 00:00:00 2001
From: Michal Simek <michal.simek@xilinx.com>
Date: Mon, 1 Jul 2013 09:32:16 +0200
Subject: [PATCH 17/20] mtd: m25p80: Fix sparse warnings

git://github.com/Xilinx/linux-xlnx.git xilinx-v14.7
commit 3f7c2d54957e950b3a36a251578185bfd374562c

- do not define device_node *np again
- is dual must be u32 because of_property_read_u32

Warning log:
drivers/mtd/devices/m25p80.c:1238:22: warning: symbol 'np' shadows an earlier one
drivers/mtd/devices/m25p80.c:1148:36: originally declared here
drivers/mtd/devices/m25p80.c:1337:23: warning: symbol 'np' shadows an earlier one
drivers/mtd/devices/m25p80.c:1148:36: originally declared here
drivers/mtd/devices/m25p80.c:1244:45: warning: incorrect type in argument 3 (different signedness)
drivers/mtd/devices/m25p80.c:1244:45:    expected unsigned int [usertype] *out_value
drivers/mtd/devices/m25p80.c:1244:45:    got int static *<noident>

Signed-off-by: Michal Simek <michal.simek@xilinx.com>
Acked-by: Suneel Garapati <suneel.garapati@xilinx.com>
Signed-off-by: Liming Wang <liming.wang@windriver.com>
---
 drivers/mtd/devices/m25p80.c |    4 +---
 1 files changed, 1 insertions(+), 3 deletions(-)

diff --git a/drivers/mtd/devices/m25p80.c b/drivers/mtd/devices/m25p80.c
index 62c177d..9c744d4 100644
--- a/drivers/mtd/devices/m25p80.c
+++ b/drivers/mtd/devices/m25p80.c
@@ -1250,9 +1250,8 @@ static int m25p_probe(struct spi_device *spi)
 
 	{
 #ifdef CONFIG_OF
-		struct device_node *np;
 		const char *comp_str;
-		static int is_dual;
+		u32 is_dual;
 		np = of_get_next_parent(spi->dev.of_node);
 		of_property_read_string(np, "compatible", &comp_str);
 		if (!strcmp(comp_str, "xlnx,ps7-qspi-1.00.a")) {
@@ -1349,7 +1348,6 @@ static int m25p_probe(struct spi_device *spi)
 		/* enable 4-byte addressing if the device exceeds 16MiB */
 		if (flash->mtd.size > 0x1000000) {
 #ifdef CONFIG_OF
-			struct device_node *np;
 			const char *comp_str;
 			np = of_get_next_parent(spi->dev.of_node);
 			of_property_read_string(np, "compatible", &comp_str);
-- 
1.7.5.4

